lib/SILGen/SILGenConvert.cpp

```diff
@@ -856,24 +856,20 @@ SILGenFunction::emitOpenExistential(
                "value mode");
         loweredOpenedType = loweredOpenedType.getAddressType();
       }
-      archetypeValue =
+      SILValue archetypeValue =
           B.createOpenExistentialAddr(loc, existentialValue.forward(*this),
                                       loweredOpenedType, allowedAccess);
-    } else {
-      archetypeValue = B.createOpenExistentialValue(
-          loc, existentialValue.forward(*this), loweredOpenedType);
-    }
-
-    if (existentialValue.hasCleanup()) {
-      // With CoW existentials we can't consume the boxed value inside of
-      // the existential. (We could only do so after a uniqueness check on
-      // the box holding the value).
-      canConsume = false;
-      enterDestroyCleanup(existentialValue.getValue());
+      if (existentialValue.hasCleanup()) {
```

Comment:
Instead of forwarding, you could just use ManagedValue::getValue(). Then you don't have to re-create the cleanup. Or you could create SILGenBuilder::createOpenExistentialAddr(...) that would just do that.